### ISSUE TITLE
New auth flow dialogs

### ISSUE THREAD

**Issue by vedmant** [ezl/dailyidea.com#871]
Here I'll add list of subtasks I need to do to complete #859 #860 #861

https://www.figma.com/file/VTAWAu8G6jtzpWYulfpw9m/Daily-Idea?node-id=340%3A1955&viewport=-1980%2C8906%2C0.5412866473197937

- [x] Update default dialog to new style
- [x] Update ask email dialog
- [x] Update ask name dialog
- [ ] ??? Move ask name dialog to after user returned and logged in to the site
- [x] Change appsync to return name in "checkEmailBelongsToExistingUser"
- [x] Create new welcome dialog
- [x] Create new resend email dialog with countdown
- [x] Refactor registerUserAndProcessIdeaAction to reuse new welcome dialog on new user instead of default dialog
- [x] Refactor auth flow for like / save / comment into single component as they are very similar
- [x] Setup separate vuetify config (optionsPath: './vuetify.options.js') and add custom icons for buttons
- [x] Refactor ActOnIdea to simplify and shorten code
- [ ] Update global loader to include appropriate message for the action (like / save / comment)
- [x] Resend dialog back to login button, what it should do?
- [x] Check possible bug with registering multiple users on email resend
- [x] Test like with existing user / new user
- [x] Test save with existing user / new user
- [x] Test comment with existing user / new user

**Comment by vedmant** [comment]
@ezl There is another a bit complicated thing with moving ask name dialog after user returned to the site. To make this work properly I'll need to make this dialog global, for example in case user will not fill the name and close site then open main page for example, it must make sure that user has some name. 

Alternatively I can add default name like "Anonymous" and if user skipped filling the name, he will have this name. 

The simplest will be just to keep ask name dialog in the beginning, then there is no additional work.

**Comment by vedmant** [comment]
@ezl Also as I understand we now remove dialogs like "You saved your first idea!", "You liked your first idea!", "We saved that idea for you!", "We liked that idea for you!" ?

**Comment by vedmant** [comment]
@Watson-B  I see there is "Back to login" button on resend dialog, what it should do?

**Comment by Watson-B** [comment]
@vedmant Great call out, that was a relic from the previous flows we had, that slipped redesign. I think changing it to "close" might make more sense. My thought process here is that if the user is resending the link, it means they already have their inbox open, so they don't need to go back to "send me to inbox" modal. @ezl @ikosovec thoughts?

![image](https://user-images.githubusercontent.com/65913664/106504603-05258780-647c-11eb-966d-5de608005002.png)

**Comment by vedmant** [comment]
I tested it and looks like it's working good, this affects like / save / comment actions. I'll merge this for testing to proceed faster with this task, also waiting on questions.

**Comment by ezl** [comment]
> @ezl There is another a bit complicated thing with moving ask name dialog after user returned to the site. To make this work properly I'll need to make this dialog global, for example in case user will not fill the name and close site then open main page, it must make sure that user has some name.
> Alternatively I can add default name like "Anonymous" and if user skipped filling the name, he will have this name.
> The simplest will be just to keep ask name dialog in the beginning, then there is no additional work.

Regarding this name thing -- Don't make it global. I understand the problem. Let me discuss with @Watson-B. As I understand it, the options (if we don't make global) are:

(A) just don't ask for it before the email -- use a placeholder name (empty string or "Anonymous"). Then when the user clicks confirmation email link and returns to the site, we can force the user to enter a name
(B) ask for the name BEFORE the email (so we can populate it before sending)

Is that correct?

---

Remove dialogs vs banners --

1. Correct, we are removing dialogs and using banners.
2. Putting at top of screen (global) is ok. @Watson-B can you update the Figma files accordingly so they'll match the implementation
3. If you'd like, we can move the dialog -> banner into another ticket. Would you prefer that @vedmant ?

---

Change button text to close (as discussed by @vedmant and @Watson-B) -- yep, I'm ok with that proposal.

**Comment by vedmant** [comment]
@ezl 

> Is that correct?

Yes, the simplest option will be ` (B) ask for the name BEFORE the email (so we can populate it before sending)` to keep it as is.

> If you'd like, we can move the dialog -> banner into another ticket. Would you prefer that @vedmant ?

Yes, lets create a ticket for that, I'd like to close this one faster. @ikosovec could you create one?

**Comment by vedmant** [comment]
@ikosovec Could you also create a ticket for "Update global loader to include appropriate message for the action (like / save / comment)". Then this issue can be closed.

**Comment by ikosovec** [comment]
On it! :) 

Dialogues -> banners: #878 
Update global loader to include appropriate message for the action (like / save / comment): #879

Closing this ticket!

**Comment by ezl** [comment]
@ikosovec I re-opened this just for the final question around when to ask for the name.

@vedmant I understand the simplest is to just ask for the name before.

What if we just don't ask for the name -- but leave it `null` before the interaction. Then when the user comes back to the site, if the name is null, we force the dialog that asks them for the name?

**Comment by vedmant** [comment]
@ezl 

> What if we just don't ask for the name -- but leave it null before the interaction. Then when the user comes back to the site, if the name is null, we force the dialog that asks them for the name?

One problem here is that before user typed his name his profile page will just not work, it has user name in the route (like /profile/sam), this might create issues. Generally should be ok if to prevent user do anything on the site unless he typed his name, but for this I'll need to create another page and middleware and redirect user there if it has no name. This still can create more issues, I'll need to handle redirects to correct pages after authentication, after redirect to set name page to redirect to initially intended page. So this might be not a small piece of work and a field for new bugs.

**Comment by ezl** [comment]
@vedmant ok understood -- in that case, let's just keep the flow where we ask for the name BEFORE the email, as is. But can you still use the styles that @Watson-B created? (so I think we'll just move the screen that asks for name from after the email to before the email).
